### ISSUE TITLE
[Question] Which Kernel options are required?

### ISSUE THREAD

**Issue by kuraga** [bluez/bluez#459]
Beyond Linux® From Scratch 11.2 [says](https://www.linuxfromscratch.org/blfs/view/11.2/general/bluez.html) that BlueZ 5.65 requires some Kernel options. Some distributions [refer to it](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-wireless/bluez/bluez-5.65-r1.ebuild?id=73b72443e97ff78a91c1bbdd26f0433e1b28c463#n81).

But I don't see the following Kernel options in any `master`'s documentation pages:

| Kernel option description                  | Kernel option              |
|--------------------------------------------|----------------------------|
| Enable timerfd() system call               | `CONFIG_TIMERFD`           |
| Enable eventfd() system call               | `CONFIG_EVENTFD`           |
| HIDP protocol support                      | `CONFIG_BT_HIDP`           |
| MD5 digest algorithm                       | `CONFIG_CRYPTO_MD5`        |
| SHA1 digest algorithm                      | `CONFIG_CRYPTO_SHA1`       |
| Diffie-Hellman algorithm | `CONFIG_CRYPTO_DH` |
| Diffie-Hellman operations on retained keys | `CONFIG_KEY_DH_OPERATIONS` |
| - | `CONFIG_CRYPTO_KPP` |

Are they required?

Thanks!

**Comment by bgix** [comment]
This won't address your Headphone issue, but the following kernel crypto functions should also be enabled:
	CONFIG_CRYPTO_USER_API
	CONFIG_CRYPTO_USER_API_AEAD
	CONFIG_CRYPTO_USER_API_HASH
	CONFIG_CRYPTO_USER_API_SKCIPHER

	CONFIG_CRYPTO_AES
	CONFIG_CRYPTO_CCM
	CONFIG_CRYPTO_AEAD
	CONFIG_CRYPTO_CMAC

**Comment by kuraga** [comment]
@bgix, thanks!

And what about (second) table's options?

P.S. Which "my Headphone issue"?

**Comment by bgix** [comment]
I'm sorry... I thought this might be related to an Audio issue that was recently discussed.  If it isn't, then my bad.

**Comment by kuraga** [comment]
@bgix , it's ok :) But what about options in the table?

**Comment by bgix** [comment]
Your second table lays outside my area of expertise.  The kernel options I added to your list have been required since Bluetooth Mesh support was added to BlueZ, but I do not know what or if the other options you list are required.

**Comment by kuraga** [comment]
@bgix , thanks! Let's ping somebody with expertise? :)

**Comment by kuraga** [comment]
@bluez, @mabhay125, @Vudentz, please?

**Comment by Vudentz** [comment]
@kuraga the config options are explained in:

https://github.com/bluez/bluez/blob/master/doc/test-runner.txt

**Comment by kuraga** [comment]
@Vudentz , thanks!

But are all parts of BlueZ tested? Because not [all options for Mesh](https://github.com/bluez/bluez/blob/5.66/README#L77) are even listed.

**Comment by Vudentz** [comment]
> @Vudentz , thanks!
> 
> But are all parts of BlueZ tested? Because not [all options for Mesh](https://github.com/bluez/bluez/blob/5.66/README#L77) are even listed.

Right, it is inconsistent since we are not listing the crypto requirements although we do seem to cover that on the tester.config itself which is what is used by our CI:

https://github.com/bluez/bluez/blob/master/doc/tester.config 

@bgix @tedd-an we should probably try to make it more consistent what are the kernel crypto requirements.
